bgp/modules/terms.py

```diff
@@ -113,3 +115,55 @@ def validate_isbn(isbn):
     def __init__(self):
         super(IsbnExtractorModule, self).__init__(self.validate_isbn)
 
+
+class PageTypeProcessor:
+
+    def __init__(self, modules):
+        self.modules = modules
+
+    '''
+    1 --> run() method sends the book(djvu.xml) file to pageProcessor.
+    2 --> pageProcessor processes the xml file and parses the pages
+    3 --> Each page is then parsed and passed into KeywordPageDetectorModule for further operations
+    4 --> KeywordPageDetectorModule looks for the keywords and returns the pages in a list called detectedPages
+          back to pageProcessor and stores them in a list called detectedPage
+    '''
+    
+    def run(self, book):
+        self.pageNo = self.pageProcessor(book.xml)
+
+    @staticmethod
+    def pageProcessor(xmlFile):
+        utf8_parser = etree.XMLParser(encoding='utf-8')
+        node = etree.fromstring(xmlFile.encode('utf-8'), parser=utf8_parser)
+        for x in node.iter('OBJECT'):
+            param = x[0].attrib['value'].split('.')[0] 
+            pageNo = param[-4:]
+            detectedPages = KeywordPageDetectorModule.keywordPageFinder(x)
+            print("\n***** Page Processor Results *****\nDetected pages for given keyword are ",detectedPages)
+            return detectedPages
+    
+    @property
+    def results(self):
+        return self.pageNo
+
+class KeywordPageDetectorModule:
+    
+    def __init__(self,keyword):
+        self.keyword = keyword
+    
+    @staticmethod  
+    def keywordPageFinder(x):
+        keyword = 'descartes' # keyword was not getting detected so given manually (Needs to be fixed)
+        detectedPages = []
+        
+        for i in x.iter('WORD'):
+            if(i.text.lower() == keyword):
+                print(i.text)
+                param = x[0].attrib['value'].split('.')[0] 
+                keywordPage = param[-4:]
+                print("Keyword",keyword, "is found at page",keywordPage) # For debugging purpose
+                detectedPages.append(keywordPage)
+        return detectedPages
```

Comment:
Instead of a debugging line, please add a test to ensure the output is what you expect it to be. :)

```suggestion
    def run(self):
        for word in x.iter('WORD'):
            if(word.text.lower() == self.keyword):
                param = x[0].attrib['value'].split('.')[0] 
                current_page = param[-4:]
                self.detected_pages.append(current_page)

    @property    
    def results(self):
        return self.detected_pages
```